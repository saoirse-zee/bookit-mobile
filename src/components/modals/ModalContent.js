import React from 'react'
import { View, Text, TouchableHighlight, StyleSheet } from 'react-native'
import colors from '../../../constants/Colors'

const ModalContent = ({
  title,
  message,
  onOkayPress,
}) => (
  <View>
    <Text style={styles.title}>{title}</Text>
    <Text>{ message }</Text>
    <View style={styles.button}>
      <TouchableHighlight onPress={onOkayPress}>
        <Text style={styles.buttonText}>Okay</Text>
      </TouchableHighlight>
    </View>
  </View>
)

export default ModalContent

const styles = StyleSheet.create({
  button: {
    padding: 20,
    alignItems: 'center',
    marginTop: 50,
  },
  buttonText: {
    fontSize: 24,
    color: colors.tintColor,
  },
  title: {
    color: colors.tintColor,
    fontSize: 30,
    fontWeight: 'bold',
    marginBottom: 50,
  },
})
