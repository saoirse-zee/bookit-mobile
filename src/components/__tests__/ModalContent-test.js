import 'react-native'
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import ModalContent from '../modals/ModalContent'

it('renders correctly', () => {
  const tree = renderer.create(<ModalContent
    title="Hey, something happened!"
    message="And I'm going to tell you about it here."
    onOkayPress={() => {}}
  />).toJSON()

  expect(tree).toMatchSnapshot()
})
