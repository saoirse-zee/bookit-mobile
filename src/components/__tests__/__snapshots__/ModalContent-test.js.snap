// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4F97FF",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 50,
      }
    }
  >
    Hey, something happened!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    And I'm going to tell you about it here.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4F97FF",
            "fontSize": 24,
          }
        }
      >
        Okay
      </Text>
    </View>
  </View>
</View>
`;
